### PR TITLE
Prevent unnecessary fm calls

### PR DESCRIPTION
**Pull request by Flix6x**

### CODE REVIEW

**Review by Flix6x — COMMENTED** [review]
@ArdJonker Nicely implemented. I only have 2 small issues.

File: flexmeasures_client.py [hunk 1]
```diff
@@ -126,8 +152,7 @@ def get_schedule(self, kwargs, **fnc_kwargs):
             params=message,
             headers={"Authorization": self.fm_token},
         )
-        self.check_deprecation_and_sunset(url, res)
```
[the remark this answers is not in the document]
Comment (ArdJonker) [reply flexmeasures_client.py:129]: OK. How can i do that? Easiest in github?

File: flexmeasures_client.py [hunk 2]
```diff
@@ -19,8 +19,9 @@ class FlexMeasuresClient(hass.Hass):
 
     # Constants
     FM_API: str
-    FM_API_VERSION: str
-    FM_QUASAR_SENSOR_ID: str
+    # FM_API_VERSION: str
+    # FM_QUASAR_SENSOR_ID: str
```
[the remark this answers is not in the document]
Comment (ArdJonker) [reply flexmeasures_client.py:23]: OK, will do

File: flexmeasures_client.py [hunk 3]
```diff
@@ -31,13 +32,21 @@ class FlexMeasuresClient(hass.Hass):
...
+    fm_busy_getting_schedule: bool
+    # Helper to prevent sending the same trigger message twice.
+    previous_trigger_message: str
 
     def initialize(self):
         self.FM_API = self.args["fm_api"]
-        self.FM_API_VERSION = self.args["fm_api_version"]
-        self.FM_QUASAR_SENSOR_ID = str(self.args["fm_quasar_sensor_id"])
+        # self.FM_API_VERSION = self.args["fm_api_version"]
+        # self.FM_QUASAR_SENSOR_ID = str(self.args["fm_quasar_sensor_id"])
```
[the remark this answers is not in the document]
Comment (ArdJonker) [reply flexmeasures_client.py:45]: OK, will do.

File: flexmeasures_client.py [hunk 4]
```diff
@@ -220,7 +249,18 @@ def trigger_schedule(self, *args, **fnc_kwargs):
                 "roundtrip-efficiency": self.WALLBOX_PLUS_CAR_ROUNDTRIP_EFFICIENCY
             }
         }
-        self.log(f"Trigger_schedule with message: {message}.")
+        # ToDo: Add min- and max-soc to request (just before roundtrip-efficiency")
+        # "soc-min": self.CAR_MIN_SOC_IN_KWH,
+        # "soc-max": self.CAR_MAX_SOC_IN_KWH,
+
```
[the remark this answers is not in the document]
Comment (ArdJonker) [reply flexmeasures_client.py:255]: True! Will remove.

File: flexmeasures_client.py [hunk 5]
```diff
@@ -251,7 +291,9 @@ def handle_response_errors(self, message, res, description, fnc, **fnc_kwargs):
             self.set_state("input_boolean.error_schedule_cannot_be_retrieved", state="off")
         else:
             self.set_state("input_boolean.error_schedule_cannot_be_retrieved", state="on")
-            self.log(f"Failed to {description} (status {res.status_code}): {res.json()} as response to {message}")
+            # Removed res.json as it crashed sometimes on (none present) json..
+            # self.log(f"Failed to {description} (status {res.status_code}): {res.json()} as response to {message}")
+            self.log(f"Failed to {description} (status {res.status_code}) as response to {message}")
```
[the remark this answers is not in the document]
Comment (ArdJonker) [reply flexmeasures_client.py:296]: True. will do

**Review by ArdJonker — APPROVED** [review]
Approve

File: flexmeasures_client.py [hunk 6]
```diff
@@ -251,7 +284,9 @@ def handle_response_errors(self, message, res, description, fnc, **fnc_kwargs):
             self.set_state("input_boolean.error_schedule_cannot_be_retrieved", state="off")
         else:
             self.set_state("input_boolean.error_schedule_cannot_be_retrieved", state="on")
-            self.log(f"Failed to {description} (status {res.status_code}): {res.json()} as response to {message}")
+            if res.json is not None:
+                self.log(f"Failed to {description} (status {res.status_code}): {res.json()} as response to {message}")
+            self.log(f"Failed to {description} (status {res.status_code}) as response to {message}")
```
Comment (Flix6x) [inline flexmeasures_client.py:289]: This should move into an `else` block to avoid logging twice.

**Pull request merged**